### PR TITLE
Minor: improve scalar functions document

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Closes #.

## Rationale for this change
When I went through https://arrow.apache.org/datafusion/user-guide/sql/scalar_functions.html, I noticed that some function descriptions were complete while others were lacking aliases.

### CODE REVIEW

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -2465,6 +2487,16 @@ Returns a slice of the array.
 array_slice(array, begin, end)
 ```
 
+#### Arguments
+
+- **array**: Array expression.
+  Can be a constant, column, or function, and any combination of array operators.
+- **begin**: Index of the first element.
```
Comment (Jefffrey) [inline docs/source/user-guide/sql/scalar_functions.md:2494]: Indicate that its 0/1 indexed?

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -2663,6 +2695,18 @@ _Alias of [array_element](#array_element)._
...
+
+_Alias of [array_has](#array_has)._
+
+### `list_has_all`
+
+_Alias of [array_has_a;;](#array_has_all)._
+
+### `list_has_any`
+
+_Alias of [array_has_any](#array_has_any)._
```
Comment (Jefffrey) [inline docs/source/user-guide/sql/scalar_functions.md:2708]: Also add links to them up here

https://github.com/apache/arrow-datafusion/blob/ff7dfc3787b46715fce2b4e5aa805bb9afb9ce3a/docs/source/user-guide/sql/scalar_functions.md?plain=1#L1731-L1791

File: docs/source/user-guide/sql/scalar_functions.md [hunk 3]
```diff
@@ -2663,6 +2695,18 @@ _Alias of [array_element](#array_element)._
 
 _Alias of [array_element](#array_element)._
 
+### `list_has`
+
+_Alias of [array_has](#array_has)._
+
+### `list_has_all`
+
+_Alias of [array_has_a;;](#array_has_all)._
```
Comment (Jefffrey) [inline docs/source/user-guide/sql/scalar_functions.md:2704]:
```suggestion
_Alias of [array_has_all](#array_has_all)._
```

typo

**Review by Jefffrey — APPROVED** [review]
Nice :+1:

File: docs/source/user-guide/sql/scalar_functions.md (hunk 2 above)
Comment (Weijun-H) [reply docs/source/user-guide/sql/scalar_functions.md]: Sorry, what do you mean?

Comment (Jefffrey) [reply docs/source/user-guide/sql/scalar_functions.md]: As in add to that list there:

```md
 - [list_has](#list_has)
 - [list_has_all](#list_has_all)
 - [list_has_any](#list_has_any) 
```

So the aliases appear in that list and are linked to

**Review by comphead — APPROVED** [review]
lgtm thanks @Weijun-H !

**Pull request merged**